server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1113,8 +1113,15 @@ private Owner setupOwner(Principal principal, String ownerKey) {
             log.warn("User {} does not have access to create consumers in org {}",
                 principal.getPrincipalName(), owner.getKey());
 
-            throw new ForbiddenException(i18n.tr("{0} is not authorized to register with " +
+            List<Owner> listOfOwner = ((UserPrincipal) principal).getOwners();
+            boolean isOwnerContained =
+                listOfOwner.stream().anyMatch(t -> t.getOwnerId().equals(owner.getOwnerId()));
```

Comment:
Couple minor things here:

- Since the principal implementation isn't entirely fixed to our own code (we may get a subclass from IT's adapters), it's a good idea to be a bit more defensive here. A null check on the list and a simple null check in the anyMatch makes this completely safe:

```boolean isOwnerContained = owners != null && owners.stream()....```

```anyMatch(t -> t != null && ...)```

- For formatting multi-line statements like this, generally break it up by operation:

```
boolean isOwnerContained = owners != null && owners.stream()
    .anyMatch(...);
```

This keeps the formatting consistent when doing method chaining, allows additional chains to be added without reformatting, and makes each individual operation clearly defined.

- Real minor/pedantic: grammatically the variable should be plural, as it contains multiple owners, e.g. listOfOwners. That can be further abbreviated to simply "owners" if the name is still available. However, keeping the "listOf" is a stylistic option if you'd prefer it.